pkg/volume/glusterfs/glusterfs.go

```diff
@@ -172,10 +172,11 @@ func (plugin *glusterfsPlugin) newMounterInternal(spec *volume.Spec, ep *v1.Endp
 	source, readOnly := plugin.getGlusterVolumeSource(spec)
 	return &glusterfsMounter{
 		glusterfs: &glusterfs{
-			volName: spec.Name(),
-			mounter: mounter,
-			pod:     pod,
-			plugin:  plugin,
+			volName:         spec.Name(),
+			mounter:         mounter,
+			pod:             pod,
+			plugin:          plugin,
+			MetricsProvider: volume.NewMetricsStatFS(plugin.host.GetPodVolumeDir(pod.UID, glusterfsPluginName, spec.Name())),
```

Comment:
Maybe we need to use `strings.EscapeQualifiedNameForDisk(glusterfsPluginName)` instead of `glusterfsPluginName`?

Reply:
+1

Reply:
Yep, addressed the comment. Thanks @mlmhl and @rootfs.